flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -488,6 +495,15 @@ private List<StreamShardHandle> getShardsOfStream(
         return shardsOfStream;
     }
 
+    private synchronized String lookupStreamArn(String streamName) throws InterruptedException {
+        if (streamNameToArnLookup.containsKey(streamName)) {
+            return streamNameToArnLookup.get(streamName);
+        }
+        String streamArn = describeStream(streamName, null).getStreamDescription().getStreamARN();
```

Comment:
It would first respect the `DESCRIBE_STREAM` restart configurations set by customer (defaults to 3 max retries). And if it fails after that, it would restart the job. Given we are caching the result, we should only see this on job startup.